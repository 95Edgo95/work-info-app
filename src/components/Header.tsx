import {attemptSignOut} from "store/user/UserActionCreators";
import {mapDispatchToProps} from "helpers/ReduxHelper";
import Button from "components/Button";
import {connect} from "react-redux";
import * as React from 'react';

import * as styles from "assets/css/header.css";
import {Link} from "react-router-dom";

interface IHeaderProps {
  attemptSignOut: () => void;
}

function Header({ attemptSignOut }: IHeaderProps) {
  return (
    <div className={styles.header}>
      <Link to="/">
        Home
      </Link>
      <Button secondary onClick={attemptSignOut}>
        Exit
      </Button>
    </div>
  );
}

const actions = {
  attemptSignOut
};

export default connect<{}, IHeaderProps>(null, mapDispatchToProps(actions))(Header);
