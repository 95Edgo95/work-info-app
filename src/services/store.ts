import createSagaMiddleware, {SagaMiddleware} from "redux-saga";
import {applyMiddleware, compose, createStore} from "redux";
import reducer from "services/reducer";
import saga from "services/saga";
import {fromJS} from "immutable";

let store: any;

const initialState: any = fromJS({});

const sagaMiddleware: SagaMiddleware<any> = createSagaMiddleware();

if (process.env && process.env.NODE_ENV !== "production") {
    const composeEnhancers: any = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    store = createStore(reducer, initialState, composeEnhancers(applyMiddleware(sagaMiddleware)));
} else {
    store = createStore(reducer, initialState, compose(applyMiddleware(sagaMiddleware)));
}

sagaMiddleware.run(saga);

export default store;
