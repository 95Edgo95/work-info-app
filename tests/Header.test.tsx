import {BrowserRouter} from "react-router-dom";
import Header from "../src/components/Header";
import store from "../__mocks__/storeMock";
import {Provider} from "react-redux";
import { mount } from "enzyme";
import * as React from "react";

describe('Header', () => {
  it('should render correctly', () => {
    const component = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
