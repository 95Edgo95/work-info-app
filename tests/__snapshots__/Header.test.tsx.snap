// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Header)>
        <Header
          attemptSignOut={[Function]}
        >
          <div>
            <Link
              replace={false}
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </Link>
            <Button
              onClick={[Function]}
              secondary={true}
            >
              <button
                className=""
                onClick={[Function]}
                type="button"
              >
                Exit
              </button>
            </Button>
          </div>
        </Header>
      </Connect(Header)>
    </Router>
  </BrowserRouter>
</Provider>
`;
